src/vm/tests/principals.rs

```diff
@@ -262,3 +265,117 @@ fn test_simple_is_standard_undefined_cases() {
         .unwrap()
     );
 }
+
+#[test]
+fn test_simple_parse_principal_version() {
+    let testnet_addr_test =
+        r#"(parse-principal version 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6)"#;
+    assert_eq!(
+        Value::UInt(26),
+        execute_against_version_and_network(
+            testnet_addr_test,
+            ClarityVersion::Clarity2,
+            StacksNetworkType::Testnet
+        )
+        .unwrap()
+        .unwrap()
+    );
+
+    let mainnet_addr_test = "(parse-principal version 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY)";
+    assert_eq!(
+        Value::UInt(22),
+        execute_against_version_and_network(
+            mainnet_addr_test,
+            ClarityVersion::Clarity2,
+            StacksNetworkType::Mainnet
+        )
+        .unwrap()
+        .unwrap()
+    );
+
+    // Note: Still works, even though the address is invalid.
```

Comment:
Why is the behavior described in this note what we want?

Reply:
I *think* the more constrained the function is, the better? I think it would be a reasonable assumption for a user that if a principal is parseable by this native function, then the principal is valid - this assumption could lead to confusion possibly.

Reply:
OK. Can do.